spec.md

```diff
@@ -0,0 +1,117 @@
+# Yearn Vesting Escrow
+
+version 0.3-dev0
+
+## Vesting Factory
+
+- the factory is initialized with a template contract of `Vesting Escrow`
+- `deploy_vesting_escrow`
+    - arguments
+        - `token: address` ERC20 funding token
+        - `recipient: address` vesting recipient
+            - can be a smart contract in v0.3
+        - `amount: uint256` total vesting amount
+        - `vesting_duration: uint256` in seconds
+        - `vesting_start: uint256 = block.timestamp` in unix epoch
+        - `cliff_duration: uint256 = 0` in seconds
+        - `open_claim: bool = True`
```

Comment:
immutable arg im guessing?

Reply:
it's the initial value that can be changed by `recipient`. we use this default to be able to create vestings towards contracts. then anyone would be able to claim it for them.